quickstarts/microsoft.kubernetes/fleet/private-hubful-fleet-deploy.json

```
@@ -0,0 +1,94 @@
+                "privateEndpointVNetPolicies": "Disabled",
+                "subnets": [
+                    {
+                        "name": "subnet",
+                        "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnet_name'), 'subnet')]",
+                        "properties": {
+                            "addressPrefix": "192.168.0.0/24",
+                            "delegations": [],
+                            "privateEndpointNetworkPolicies": "Disabled",
+                            "privateLinkServiceNetworkPolicies": "Enabled"
+                        },
+                        "type": "Microsoft.Network/virtualNetworks/subnets"
+                    }
+                ],
+                "virtualNetworkPeerings": [],
+                "enableDdosProtection": false
+            }
+        },
+        {
+            "type": "Microsoft.ContainerService/fleets",
+            "apiVersion": "2025-03-01",
+            "name": "[parameters('fleets_priv_fleet_name')]",
+            "location": "eastus",
+            "dependsOn": [
+                "[guid(resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnet_name'), 'subnet'), parameters('fleets_sp_object_id'))]"
+            ],
+            "properties": {
+                "hubProfile": {
+                    "agentProfile": {
+                        "subnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnet_name'), 'subnet')]"
+                    },
+                    "apiServerAccessProfile": {
+                        "enablePrivateCluster": true,
+                        "enableVnetIntegration": false
+                    }
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks/subnets",
+            "apiVersion": "2024-05-01",
+            "name": "[concat(parameters('vnet_name'), '/subnet')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/virtualNetworks', parameters('vnet_name'))]"
+            ],
+            "properties": {
+                "addressPrefix": "192.168.0.0/24",
+                "delegations": [],
+                "privateEndpointNetworkPolicies": "Disabled",
+                "privateLinkServiceNetworkPolicies": "Enabled"
+            }
+        },
+        {
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2022-04-01",
+            "name": "[guid(resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnet_name'), 'subnet'), parameters('fleets_sp_object_id'))]",
+            "scope": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnet_name'), 'subnet')]",
+            "properties": {
+                "roleDefinitionId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4d97b98b-1d4f-4787-a291-c67834d212e7')]",
+                "principalId": "[parameters('fleets_sp_object_id')]"
+            },
```

nit, include principalType in case the RP has just been registered and the SP has only just been created and isn't well cached yet

```suggestion
            "properties": {
                "roleDefinitionId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4d97b98b-1d4f-4787-a291-c67834d212e7')]",
                "principalType": "ServicePrincipal",
                "principalId": "[parameters('fleets_sp_object_id')]"
            },
```